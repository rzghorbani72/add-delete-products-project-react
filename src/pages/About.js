import React,{ PureComponent } from 'react';
import Header from "../components/header";

class About extends PureComponent {
	render () {
		return (
			<React.Fragment>
				<Header/>
			<div id="About" className="pageWrapper col-12 row">
					<div className={`col-lg-3 col-sm-12`}/>
						<div className={`col-lg-6 col-sm-12 cardWrapper`}>
							<div>
								<h2>درباره ما</h2>
							</div>
							<div className={`desctiption`}>
									<span>لورم ایپسوم یا طرح‌نما (به انگلیسی: Lorem ipsum) به متنی آزمایشی و بی‌معنی در صنعت چاپ، صفحه‌آرایی و طراحی گرافیک گفته می‌شود. طراح گرافیک از این متن به عنوان عنصری از ترکیب بندی برای پر کردن صفحه و ارایه اولیه شکل ظاهری و کلی طرح سفارش گرفته شده استفاده می نماید، تا از نظر گرافیکی نشانگر چگونگی نوع و اندازه فونت و ظاهر متن باشد. معمولا طراحان گرافیک برای صفحه‌آرایی، نخست از متن‌های آزمایشی و بی‌معنی استفاده می‌کنند تا صرفا به مشتری یا صاحب کار خود نشان دهند که صفحه طراحی یا صفحه بندی شده بعد از اینکه متن در آن قرار گیرد چگونه به نظر می‌رسد و قلم‌ها و اندازه‌بندی‌ها چگونه در نظر گرفته شده‌است. از آنجایی که طراحان عموما نویسنده متن نیستند و وظیفه رعایت حق تکثیر متون را ندارند و در همان حال کار آنها به نوعی وابسته به متن می‌باشد آنها با استفاده از محتویات ساختگی، صفحه گرافیکی خود را صفحه‌آرایی می‌کنند تا مرحله طراحی و صفحه‌بندی را به پایان برند.
توسط فرم زیر می توانید متن ساختگی مورد نظر خود را در واحدهای کاراکتر، کلمه و پاراگراف تولید کنید، سپس آنرا کپی کنید و در کار مورد نظر خود قرار دهید.
زبان مورد نظر خود را انتخاب کنید ...
									</span>
								</div>
						</div>
					<div className={`col-lg-3 col-sm-12`}/>
				</div>
			</React.Fragment>
		);
	}
}
export default About;
