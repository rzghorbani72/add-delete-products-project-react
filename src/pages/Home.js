import React from 'react';
import {connect} from 'react-redux';
import {isEmpty} from 'lodash';
import {getPosts, updatePosts} from "../redux/actions/action";
import {GET_POSTS} from "../redux/actions/type";

//import Header from "../components/header";

export class Home extends React.Component {
	componentDidMount() {
		this.props.getPosts(GET_POSTS);
	}
	
	render() {
		const {postsInfo} = this.props;
		const posts = isEmpty(postsInfo) ? [] : postsInfo.posts;
		return (
			<React.Fragment>
				{/*<Header/>*/}
				<div id="Index" className="col-12 row pageWrapper">
					<div className={`resetBtn`}>
						<button onClick={() => this.props.getPosts()} className="alert alert-success">بارگیری مجدد آیتم ها</button>
					</div>
					<div className={`container col-12 row`}>
						{isEmpty(posts) ? '' :
							posts.map((info, i) => (
								<div key={i} className={`col-lg-3 col-md-4 col-sm-6 col-xs-12 cardWrapper`}>
									<div className={`card mx-auto`}>
										<img src={info.image} alt={info.title}/>
										<div className={`itemName`}><a href={`/post/${info.id}`}>{info.title}</a></div>
										<div className={`col-12 row`}>
											<div className={`col-lg-6 col-sm-12 alert alert-danger actionBtn`}>
												<button className={`alert-danger`} onClick={() => this.props.updatePosts(info.id)}>حذف</button>
											</div>
											<div className={`col-lg-6 col-sm-12 alert alert-info actionBtn`}>
												<a href={`/post/${info.id}`}>
													<button className={`alert-info`}>مشاهده جزییات</button>
												</a>
											</div>
										</div>
									</div>
								</div>
							))
						}
					</div>
				</div>
			</React.Fragment>
		);
	}
}

const mapStateToProps = state => ({
	...state,
	output: state.output
});
export default connect(mapStateToProps, {
	getPosts,
	updatePosts
})(Home);
