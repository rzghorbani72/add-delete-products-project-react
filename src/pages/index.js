import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import {isEmpty} from 'lodash';
import { updatePosts,getPosts} from "../redux/actions/action";
import {GET_POSTS} from "../redux/actions/type";

class Index extends React.Component {
	componentDidMount() {
		this.props.getPosts(GET_POSTS);
	}
	render () {
		const {postsInfo} = this.props;
		const posts = postsInfo.posts;
		return (
			<div id="Index" className="container-fluid">
				<button onClick={()=> this.props.getPosts()} className="alert alert-light">بارگیری مجدد آیتم ها</button>
				{isEmpty(posts) ? '' :
					posts.map((item,i)=>(
						<div key={i}>
							<a href={`/post/${item.id}`}>{item.title}</a>
							<button onClick={()=> this.props.updatePosts(item.id)}>delete</button>
						</div>
					))
				}
			</div>
		);
	}
}
const mapStateToProps = state => state;
export default connect(mapStateToProps,{getPosts,updatePosts})(withRouter(Index));
