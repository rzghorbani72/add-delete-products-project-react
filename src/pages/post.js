import React,{ PureComponent } from 'react';
import { withRouter } from 'react-router-dom';
import {connect} from "react-redux";
import {getPosts, updatePosts} from "../redux/actions/action";
import _ from 'lodash';
import {get} from './../static/defaultPosts';

class Post extends PureComponent {
	constructor (props) {
		super(props);
		this.state = {
			postDetails: {}
		};
		if(_.isEmpty(this.props.match.params.id)) this.props.history.push(`/`);
	}
	componentDidMount () {
		const {match} = this.props;
		get().then(data=>{
			 const details =_.find(data,{id:Number(match.params.id)});
			 _.isEmpty(details) ? this.props.history.push(`/`) : this.setState({postDetails:details});
		}).catch(err=>{
			console.log(err);
		})
	};
	render () {
		const {postDetails} = this.state;
		return (
			<div id="Post" className="container-fluid">
				{_.isEmpty(postDetails) ? '' :
					<div>
						<img src={postDetails.image} width={200} height={100} />
						<div>
							<span>title:</span>
							<span>{postDetails.title}</span>
						</div>
						<div>
							<span>averagePrice</span>
							<span>{postDetails.averagePrice}</span>
						</div>
					</div>
				}
			</div>
		);
	}
}
const mapStateToProps = state => state;
export default connect(mapStateToProps,{getPosts,updatePosts})(withRouter(Post));
