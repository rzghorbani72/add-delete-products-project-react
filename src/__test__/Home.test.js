import React from 'react';
import renderer from 'react-test-renderer';
import {Home} from '../pages/Home';
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// Actions to be tested
// import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore({});

Enzyme.configure({adapter: new Adapter()});
const initialProps = {
	getPosts: jest.fn()
};
describe('Home Page', () => {
	let HomePage;
	beforeAll(() => {
		HomePage = shallow(<Home {...initialProps}/>);
	});
	it('capture Snapshot of Home', () => {
		const renderedValue = renderer.create(<Home {...initialProps}/>)
			.toJSON();
		expect(renderedValue)
			.toMatchSnapshot();
	});
	it('reset button content', () => {
		const HomePage = shallow(<Home {...initialProps}/>);
		expect(HomePage.find('.alert-success')
			.text())
			.toEqual('بارگیری مجدد آیتم ها');
	});
});
