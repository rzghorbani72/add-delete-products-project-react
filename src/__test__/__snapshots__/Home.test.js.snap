// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page capture Snapshot of Home 1`] = `
<div
  className="col-12 row pageWrapper"
  id="Index"
>
  <div
    className="resetBtn"
  >
    <button
      className="alert alert-success"
      onClick={[Function]}
    >
      بارگیری مجدد آیتم ها
    </button>
  </div>
  <div
    className="container col-12 row"
  >
    
  </div>
</div>
`;
